main.js

```diff
@@ -0,0 +1,142 @@
+// The app collection
+const AppCollection = Backbone.Collection.extend({
+  // grabs the url thats passed down from the input
+  url() {
+    return this.models[0].attributes.url;
+  },
+  // on the first instance on the add event do the api fetch
+  initialize() {
+    this.on('add', function (model) {
+      model.fetch();
+    });
+  },
+});
+
+// the App Model that references the collection
+const AppModel = Backbone.Model.extend({
+  defaults() {
+    return {
+      videos: new AppCollection(),
+    };
+  },
+});
+
+// the App View that grabs the input and renders the clickable images
+const AppView = Backbone.View.extend({
+  el: 'body',
+
+  // handlebars template
+  template: Handlebars.compile($('#vid-list').html()),
+
+  // event for the click on the input button invokes the add to model
+  events: {
+    'click .search-btn': 'videoSearchAdd',
+  },
+
+  // on instance grab the input id & listene to changes on the videos model and call renderThumbnails
+  initialize() {
+    this.$searchInput = this.$('#search-input');
+    // this.$vidView = $('.vid-container');?
+    this.listenTo(this.model.get('videos'), 'change', this.renderThumbnails);
+  },
+
+  // this function grabs the value of input adds it to the model so it could be grabbed by the url search
+  videoSearchAdd() {
+    const inputValue = this.$searchInput.val();
+
+    const urlInput = `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=${inputValue}&type=video&videoEmbeddable=true&key=AIzaSyCn7T_poRhx_PIw7R7PrfqQXYuRJH8RQjw`;
+    this.model.get('videos').add({
+      url: urlInput,
+    });
+  },
+
+  //  renders the the clickable images of the videos on the right side of page
+  renderThumbnails() {
+    const mapped = this.model.attributes.videos.models[0].attributes.items.map(
```

Comment:
This is all kinds of anti-pattern. Never access `attributes` directly and why would we find all that we need inside of the first model?

The only reason you had to write this is because all of these videos should be living on the collection, instead of copied onto each "model" inside of the videos colleciton.